roles/openshift_on_openstack_scaleup/tasks/main.yml

```diff
@@ -1,71 +1,137 @@
-    - "cd openshift-ansible ; git describe"
-    - "rpm -q atomic-openshift-utils openshift-ansible"
-  changed_when: false
-  ignore_errors: true
+    difference: "{{ openshift_node_target | int - current_node_count | int }}"
 
-# Print the debug information out.
-- name: Printing the debug information to the screen
-  debug:
-    msg: "{{ item['stdout_lines'] }}"
-  with_items: "{{ debug_information['results'] }}"
+# The current node count is the starting node number.
+- name: Setting the start number
+  set_fact:
+    start_node_number: "{{ current_node_count }}"
 
-# Set the path for the OpenStack scaleup log file.
-- name: Creating the OpenStack scaleup log file variable
+# The current node count + difference - 1 is the ending node number.
+- name: Setting the end number
   set_fact:
-    openstack_scaleup_log: "{{ ansible_user_dir }}/openstack_scaleup.log"
+    end_node_number: "{{ current_node_count | int + difference | int - 1 }}"
+
+# Create the new nodes inventory file name
+- name: Creating the file name for the new nodes inventory
+  set_fact:
+    new_nodes_inventory: "{{ ansible_user_dir }}/new_nodes_{{ start_node_number }}-{{ end_node_number }}_inventory"
+
+# Search for regular expressions and replace in file.
+- name: Replacing specific lines in {{ all_yml_path }}
+  lineinfile:
+    path: "{{ all_yml_path }}"
+    regexp: "{{ item['find'] }}"
+    line: "{{ item['replace'] }}"
+  with_items:
+    - { find: "^#?openshift_openstack_num_nodes.*", replace: "openshift_openstack_num_nodes: {{ openshift_node_target }}" }
+
+# Make sure that openshift_release does not have a 'v' in the value due to an inconsistent usage in openshift-ansible.
+# https://github.com/openshift/openshift-ansible/issues/6471
+- name: Replacing specific lines in {{ osev3_yml_path }}
+  lineinfile:
+    path: "{{ osev3_yml_path }}"
+    regexp: "{{ item['find'] }}"
+    line: "{{ item['replace'] }}"
+  with_items:
+    - { find: "^#?openshift_release.*", replace: "openshift_release: \"{{ ocp_major_minor }}\"" }
 
 # Run the Ansible playbook that creates the OpenStack resources.
 - name: Creating the OpenStack resources
-  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_openstack_dir }}/openshift-cluster/provision.yml -e openshift_repos_enable_testing=true 2>&1 > {{ openstack_scaleup_log }}"
+#  shell: "{{ ansible_playbook }} -vvv --user openshift -i inventory -i {{ inventory_py }} {{ openshift_cluster_directory }}/provision.yml -e openshift_repos_enable_testing=true 2>&1 > {{ openstack_scaleup_log }}"
+  shell: >
+    {{ ansible_playbook }} -vvv
+    --user openshift
+    -i inventory
+    -i {{ inventory_py }}
+    {{ openshift_cluster_directory }}/provision.yml 2>&1 >> {{ openstack_scaleup_log }}
+  args:
+    # Use bash to get the posix style redirects.
+    executable: /bin/bash
+
+# Get the new OpenStack VMs by name.
+- name: Getting the new OpenStack VM network information by name
```

Comment:
This will be quite time-consuming.  At some point, we'll need to create the inventory faster.

Reply:
agreed. I had thought about getting listing of all the name and networks and parsing that list, but the Ansible task would have to loop over two different lists of different sizes in the same task (`with_items` from the output result and `with_sequence` to only add the new nodes).